### PR TITLE
Format source code

### PR DESCRIPTION
**Pull request by bjoernhaeuser**
I want to make the source code a bit nicer to read and mor consistent. I was not able to find a general jenkins plugins style guide so I just applied the general one from intellij.

@samrocketman what do you think?

### CODE REVIEW

**Comment by bjoernhaeuser** [comment]
If we want to go down this road I would apply the changes to all code.

**Comment by samrocketman** [comment]
I'm not against changing code style.  However, if we're really going forward with this we're missing two things:

1. Documentation:  like `STYLE_GUIDE.md` and a reference to the style guiding in `CONTRIBUTING.md`
2. Automated code style validation which runs with the tests.  People suck at strict code style so we really need a pass/fail test for code style.

Otherwise, I'm not sure it's worth opening this can of worms to make tons of style changes.  The docs serve as a reference when people contest the style and the automated validation serve to force people to correct style before their change can be merged.

**Comment by bjoernhaeuser** [comment]
@samrocketman agreed!

What I did:

* Added checkstyle maven plugin
* Added `checkstyle.xml` and adapted a bit to the current project
* Work on the violations bring the amount of violations to zero
* Added a `STYLE_GUIDE.md` which is now rather short on context

**Comment by samrocketman** [comment]
Something to be aware of: merging this will cause merge conflicts on a lot of open PRs.  We have to be willing to help (even resolve) with some of these conflicts.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbTriggerBackwardsCompatible.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
+// CHECKSTYLE:OFF
```
[the remark this answers is not in the document]
Comment (bjoernhaeuser) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbTriggerBackwardsCompatible.java]: It has a ton of violations which are hard to fix. This class looks like something we should remove soonish (next major version maybe).

**Comment by bjoernhaeuser** [comment]
I agree with everything you said.

**Comment by samrocketman** [comment]
Great job.  I know this kind of stuff is a lot of work.  A huge attention to detail.

**Pull request merged**